### DISCUSSION TITLE
Can FD use io_uring for faster performance?

### DISCUSSION THREAD

**Discussion by ScottRosenberg2** [sharkdp/fd#1312] [Ideas]
I'm not sure if it actually would help, but could this use io_uring for faster performance, by using less syscalls?

There are some rather impressive benchmarks of io performance using io_uring, so I was curious if it could help in this case.

**Comment by tavianator** [comment]
There is no io_uring operation for reading directories yet.  I have experimented with a setup that uses io_uring to open multiple dirs at once and distributes `getdents()` to a thread pool, but io_uring doesn't seem to help over just doing `open()` in the same thread pool.

**Reply by tavianator** [reply]
I see a new version of the getdents() patch has been posted: https://lore.kernel.org/io-uring/20230422-uring-getdents-v1-0-14c1db36e98c@codewreck.org/T/

**Reply by ScottRosenberg2** [reply]
From what I understand, io_uring can be used in a threadpool, but you basically have to have an instance of the circular buffer for each thread
